### PR TITLE
Use correct mesh and jacobian for integrals with subcell

### PR DESCRIPTION
**Pull request by ffoucart**
## Proposed changes

Modify ObserveVolumeIntegrals so that it uses the Dg or Subcell mesh as appropriate.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by knelli2 — APPROVED** [review]

**Review by knelli2 — CHANGES_REQUESTED** [review]
I don't think all executables have the `Events::Tags::ObserverXXX` tags implemented. What if you didn't put these tags in the `argument_tags` list, but instead only had the observation box. Then inside the function, check if the observation box has the event tags; if it does, get that, if not, use the DG one?

**Comment by ffoucart** [comment]
I guess it's that or updating all relevant executables... as this is an observer, ideally any executable that calls it could presumably use the more generic Events::Tag::XXX tags. I'm happy to go either way.

**Comment by knelli2** [comment]
@ffoucart I have no preference. Whichever is less work for you.

Edit: Actually, if you change all the executables, that makes life easier for #5053.

File: src/ParallelAlgorithms/Events/ObserveVolumeIntegrals.hpp [hunk 1]
```diff
@@ -120,8 +121,9 @@ class ObserveVolumeIntegrals<
 
   using argument_tags = tmpl::list<
       ::Tags::ObservationBox, ObservationValueTag,
-      domain::Tags::Mesh<VolumeDim>,
-      domain::Tags::DetInvJacobian<Frame::ElementLogical, Frame::Inertial>,
+      ::Events::Tags::ObserverMesh<VolumeDim>,
+      ::Events::Tags::ObserverDetInvJacobian<Frame::ElementLogical,
+                                           Frame::Inertial>,
```
Comment (nilsvu) [inline src/ParallelAlgorithms/Events/ObserveVolumeIntegrals.hpp]: Could you run clang-format over this? What I usually do is run git-clang-format when I have staged changes, or something like `git-clang-format HEAD~1` to format the last commit.

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Comment by nilsvu** [comment]
There's a CI error. I think you have to add the ObserverDetInvJac tag to some executables.

**Comment by ffoucart** [comment]
Yes; that's related to Kyle's comment, but I have not had the chance to fix it yet (yesterday was a holiday in the US)

**Review by nilsvu — APPROVED** [review]

**Pull request merged**